### PR TITLE
Rails 5.0 support

### PR DESCRIPTION
**Pull request by chancefeick**
Add support for Rails 5.0. Introduce the following changes:

- Add Ruby 2.4 to build matrix for Rails 5 only
- Use `VersionSpecifier` to ensure CI passes for all supported Rails versions
- Update job subscriber to use `QueueAdapter` class as tag value in Rails >= 5.0

Active Support instrumentation API changed the default event payload value to `QueueAdapter` object processing the job in Rails 5.0, the event payload was already a class in Rails 4.2.

### CODE REVIEW

File: lib/librato/rails/subscribers/job.rb [hunk 1]
```diff
@@ -9,11 +9,17 @@ module Subscribers
...
 
           event = ActiveSupport::Notifications::Event.new(*args)
+
           tags = {
-            adapter: event.payload[:adapter].to_s.demodulize.underscore,
+            adapter: event.payload[:adapter].class.to_s.demodulize.underscore,
             job: event.payload[:job].class.to_s.demodulize.underscore
           }
 
+          Librato::Rails::VersionSpecifier.supported(max: '4.2') do
```
Comment (nextmat) [inline lib/librato/rails/subscribers/job.rb]: Since this is already inside the `Librato::Rails` namespace, this can just be `VersionSpecifier` unless you are intentionally being more explicit.

File: test/dummy/app/controllers/cache_controller.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class CacheController < ApplicationController
-  before_filter :instrument_caching
+  Librato::Rails::VersionSpecifier.supported(max: '4.1') { before_filter :instrument_caching }
```
Comment (nextmat) [inline test/dummy/app/controllers/cache_controller.rb]: why is this needed?

Comment (chancefeick) [inline test/dummy/app/controllers/cache_controller.rb]: It was already deprecated and finally removed in Rails 5:

```
DEPRECATION WARNING: ActiveSupport::Cache.instrument= is deprecated and will be removed in Rails 5. Instrumentation is now always on so you can safely stop using it.
```

Comment (nextmat) [inline test/dummy/app/controllers/cache_controller.rb]: Ah, makes sense - maybe add a comment to this effect for future selves? Would be nice to make it clear that it isn't needed after 4.1 because it is always on after that, we aren't actually removing something.

Comment (chancefeick) [inline test/dummy/app/controllers/cache_controller.rb]: Yeah, I was hoping that using a version specifier would convey that it is no longer supported, but your suggestion definitely makes sense so we know to remove it when/if we drop 4.1 support in the future. Will add a comment.

**Comment by chancefeick** [comment]
@nextmat updated based on feedback, also added some more detailed comments

File: test/dummy/app/controllers/cache_controller.rb [hunk 3]
```diff
@@ -1,5 +1,9 @@
 class CacheController < ApplicationController
-  before_filter :instrument_caching
+  Librato::Rails::VersionSpecifier.supported(max: '4.1') do
+    # ActiveSupport::Cache.instrument= was removed in Rails 5.
```
Comment (nextmat) [inline test/dummy/app/controllers/cache_controller.rb]: You might change this to deprecated in 4.2 so that the version you are limiting on makes sense in context.

Comment (chancefeick) [inline test/dummy/app/controllers/cache_controller.rb]: Updated 👍
